### DISCUSSION TITLE
BAM Converter - Sort Palette?

### DISCUSSION THREAD

**Discussion by TheArtisanBG** [NearInfinityBrowser/NearInfinity#165] [Q&A] [answered]
This is rather sudden but I was wondering whether the palette sorting function on the BAM Converter is functioning as intended. The problem I've been facing involves some custom animations I'm adding in which are not sorting the colors in the palette no matter what I change the setting to - this is causing major issues as it means the black palette color for shadows isn't being sorted to the position it's supposed to be in leading to opaque shadows... when I look at some of the first animations I made the palettes are sorted into a gradient which  though this may have happened somewhere else that I've forgot about. I'm really clueless on how the program functions, so it's possible I've horribly misunderstood something - for all I know the problem could actually be with GIMP and I'm looking at the wrong thing altogether.

**Comment by Argent77** [comment, marked as answer]
Palette sorting is only applied when a new palette is generated internally. That usually happens when you import or add BAM frames, but not if the imported frames already come with a prepared palette that doesn't require post-processing.

It is also possible to manually arrange individual colors in the palette dialog by dragging palette entries with the mouse to their desired position.

**Comment by TheArtisanBG** [comment]
Thanks. It's been too long since I tried making custom animations again - I figured out what I did before that I wasn't doing this time, so it wasn't a NearInfinity issue. My bad.

On another note... is there any chance that you could implement more keyboard controls, such as copy/pasting effects with Ctrl+C/V hotkeys, just for efficiency's sake? I hope it wouldn't be too much of a problem.

**Comment by Argent77** [comment]
> On another note... is there any chance that you could implement more keyboard controls, such as copy/pasting effects with Ctrl+C/V hotkeys, just for efficiency's sake? I hope it wouldn't be too much of a problem.

I'll see what I can do. The clipboard for copy/pasted data of structured resources is a custom implementation and works independently from the system clipboard. I'd have to use different shortcuts to avoid conflicts.
